doc/src/sgml/xlstproc_depfile_wrapper.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+import os
+import shutil
+import subprocess
+import sys
+import argparse
+from unittest import result
+
+parser = argparse.ArgumentParser(
+    description='generate dependency file for docs')
+
+parser.add_argument('--basename', type=str, required=False, nargs='+')
+parser.add_argument('--depfile', type=str, required=False)
+parser.add_argument('-i', '--input', type=str, nargs='+')
+parser.add_argument('-o', '--output', type=str, required=True)
+parser.add_argument('--path', type=str, required=True)
+parser.add_argument('--xsltproc', type=str, required=True)
+
+# since both flags and inputs can have multiple arguments
+# and also flags can have arguments starts with `--`,
+# flags parsed as a remaining args
+args, flags = parser.parse_known_args()
```

Comment:
I don't like the parse_known_args() approach, it's easy to have conflicts between the binaries etc. I think it's better to use an explicit separator for arguments intended for the wrapper and arguments intended for xsltproc itself. I changed this for e.g. src/tools/pgflex (check the flex_flags argument).